Sources/OktaLogger/Analytics/Analytics.swift

```diff
@@ -0,0 +1,214 @@
+        providers.forEach { Self.providers.removeValue(forKey: $0) }
+    }
+
+    /**
+     Track event to all providers provided by client
+
+    - Parameters:
+        - event: `AnalyticsEvent` or `AnalyticsEventProtocol` provided by client
+        - logLevel: default to `.analytics` or log different leven provided by caller
+    */
+    public static func trackEvent(_ event: AnalyticsEventProtocol, logLevel: OktaLoggerLogLevel = .analytics) {
+        providers.forEach { trackEvent(event, to: $1, logLevel: logLevel) }
+    }
+
+    /**
+     Remove providers from collection
+
+    - Parameters:
+        - provider: `Provider`s provided by client, if exists
+    */
+    public static func purge() {
+        providers.removeAll()
+    }
+}
+
+private extension Analytics {
+    static func trackEvent(_ event: AnalyticsEventProtocol, to provider: Provider, logLevel: OktaLoggerLogLevel = .analytics) {
+        guard let provider = providers[provider] else {
+            return
+        }
+        if provider.isTrackingEnabled {
+            event.properties += provider.defaultProperties
+            trackEvent(event, to: provider, logLevel: logLevel)
+        }
+    }
+
+    static func trackEvent(_ event: AnalyticsEventProtocol, to providers: [Provider], logLevel: OktaLoggerLogLevel = .analytics) {
+        providers.forEach { provider in
+            guard let provider = Self.providers[provider] else {
+                return
+            }
+            if provider.isTrackingEnabled {
+                event.properties += provider.defaultProperties
+                trackEvent(event, to: provider, logLevel: logLevel)
+            }
+        }
+    }
+
+    static func trackEvent(_ event: AnalyticsEventProtocol, to providers: [AnalyticsProviderProtocol], logLevel: OktaLoggerLogLevel = .analytics) {
+        providers.forEach { provider in
+            if provider.isTrackingEnabled {
+                event.properties += provider.defaultProperties
+                trackEvent(event, to: provider, logLevel: logLevel)
+            }
+        }
+    }
+
+    static func trackEvent(_ event: AnalyticsEventProtocol, to provider: AnalyticsProviderProtocol, logLevel: OktaLoggerLogLevel = .analytics) {
+        if provider.isTrackingEnabled {
+            event.properties += provider.defaultProperties
+            provider.logger.log(level: logLevel, eventName: event.name, message: nil, properties: event.properties)
+        }
+
+        switch provider.name {
+        case .appCenter:
+            switch logLevel {
+            case .error:
+                AppCenterAnalytics.Analytics.trackEvent(event.name, withProperties: event.properties, flags: .critical)
+            default:
+                AppCenterAnalytics.Analytics.trackEvent(event.name, withProperties: event.properties)
+            }
+        }
+    }
+}
+
+extension Dictionary {
+    static func += (lhs: inout Self, rhs: Self) {
+        lhs.merge(rhs) { current, _ in current }
```

Comment:
Not sure `+=` is more readable/better than just `merge`

Reply:
changed with merge instead of operator overloading